benchmark/GasCostBenchmark.js

```diff
@@ -0,0 +1,298 @@
+const DocumentStore = artifacts.require("./DocumentStore.sol");
+const DocumentStoreWithRevokeReasons = artifacts.require("./DocumentStoreWithRevokeReasons.sol");
+const ProxyFactory = artifacts.require("./ProxyFactory.sol");
+const BaseAdminUpgradeabilityProxy = artifacts.require("./BaseAdminUpgradeabilityProxy.sol");
+
+DocumentStore.numberFormat = "String";
+const {generateHashes} = require("../scripts/generateHashes");
+
+const initializeAbi = {
+  constant: false,
+  inputs: [
+    {
+      internalType: "string",
+      name: "_name",
+      type: "string"
+    },
+    {
+      internalType: "address",
+      name: "owner",
+      type: "address"
+    }
+  ],
+  name: "initialize",
+  outputs: [],
+  payable: false,
+  stateMutability: "nonpayable",
+  type: "function"
+};
+
+const STORE_NAME = "THE_STORE_NAME";
+let lastIssuedHash = "0x3a267813bea8120f55a7b9ca814c34dd89f237502544d7c75dfd709a659f6432";
+const randomHashes = num => {
+  const generated = generateHashes(num, lastIssuedHash);
+  lastIssuedHash = generated[generated.length - 1];
+  return generated;
+};
+const randomHash = () => randomHashes(1)[0];
+
+describe("Gas Cost Benchmarks", () => {
+  const gasRecords = [];
+  const recordGasCost = (contract, context, gas) => {
+    gasRecords.push({
+      contract,
+      context,
+      gas
+    });
+  };
+  after(() => {
+    // eslint-disable-next-line no-console
+    console.table(gasRecords);
+  });
+
+  let staticDocumentStoreInstance;
+  let staticProxyFactoryInstance;
+  let staticDocumentStoreWithRevokeReasons;
+
+  before(async () => {
+    staticDocumentStoreInstance = await DocumentStore.new();
+    staticProxyFactoryInstance = await ProxyFactory.new();
+    staticDocumentStoreWithRevokeReasons = await DocumentStoreWithRevokeReasons.new();
+  });
+
+  contract("DocumentStore", accounts => {
+    let documentStoreInstance;
+    it("deploy", async () => {
+      const deployment = await DocumentStore.new();
+      documentStoreInstance = deployment;
+      const receipt = await web3.eth.getTransactionReceipt(deployment.transactionHash);
+      recordGasCost("DocumentStore", "deployment", receipt.cumulativeGasUsed);
+    });
+    it("initialize", async () => {
+      const {receipt} = await documentStoreInstance.initialize(STORE_NAME, accounts[0]);
+      recordGasCost("DocumentStore", "initialize", receipt.cumulativeGasUsed);
+    });
+    it("transferOwnership", async () => {
+      const {receipt} = await documentStoreInstance.transferOwnership(accounts[1]);
+      recordGasCost("DocumentStore", "transferOwnership", receipt.cumulativeGasUsed);
+      await documentStoreInstance.transferOwnership(accounts[0], {from: accounts[1]});
```

Comment:
no record needed ?

Reply:
Replaced this to make this an independent test. It was there previously to make sure the state of the documentStoreInstance is the same as before the test ran.